src/backend/postmaster/postmaster.c

```diff
@@ -3873,11 +3912,37 @@ HandleChildCrash(int pid, int exitstatus, const char *procname)
 		BgWriterPID = 0;
 	else if (BgWriterPID != 0 && take_action)
 	{
-		ereport(DEBUG2,
-				(errmsg_internal("sending %s to process %d",
-								 (SendStop ? "SIGSTOP" : "SIGQUIT"),
-								 (int) BgWriterPID)));
-		signal_child(BgWriterPID, (SendStop ? SIGSTOP : SIGQUIT));
+		if (delay_bg_writer_termination_sec == 0)
+		{
+			ereport(DEBUG2,
+					(errmsg_internal("sending %s to process %d",
+									 (SendStop ? "SIGSTOP" : "SIGQUIT"),
+									 (int) BgWriterPID)));
+			signal_child(BgWriterPID, (SendStop ? SIGSTOP : SIGQUIT));
+		}
+		else
+		{
+			/*
+			 * Code below is only for test purposes and should not be invoked
+			 * under normal conditions. It will delay the termination of the
+			 * background writer process.
+			 *
+			 * In order to delay the termination of the background writer
+			 * process - send SIGSTOP to the bg writer process and start a
+			 * timer that will wake up and then immediately terminate it.
+			 */
+			ereport(DEBUG2,
+					(errmsg_internal("sending SIGSTOP to process %d and starting timer",
+									 (int) BgWriterPID)));
+			signal_child(BgWriterPID, SIGSTOP);
+
+			static TimeoutId id = 0;
```

Comment:
> Definition should be moved to the start of the block.
 
Yes, moved it.

> Why does it need to be static?

It has to be static, because we need to call `RegisterTimeout` only once before first call of `enable_timeout_after`.
And on subsequent calls we reuse the timer id. If `static` is removed, soon we will run out of available timers.